### PR TITLE
Adding AIX classes, tests, and targetted ssh testing

### PR DESCRIPTION
**Pull request by foobarbam**
```
rake test:ssh[user@server]
sh -c bin/inspec exec /home/foobarbam/src/gems/inspec/test/integration/test/integration/default/* --sudo -t ssh://user@server --key_files=/home/foobarbam/.ssh/id_rsa
"You are currently running on OS family: aix, OS release: 6.1"
.............................................................................................................

Finished in 21.65 seconds (files took 19.21 seconds to load)
109 examples, 0 failures

```

### CODE REVIEW

**Comment by foobarbam** [comment]
I can fix this, but I don't have access to any OSX systems to grab some command output.

Could someone `lsof -nP -i -FpctPn` and get me the output?

**Comment by chris-rock** [comment]
@foobarbam huge addition. Thank you very much for this great Christmas present. I am going to test this PR as my my first task next week.

File: test/integration/test/integration/default/package_spec.rb [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
   describe package('curl') do
     it { should be_installed }
   end
-
-  describe package('nginx') do
-    it { should_not be_installed }
+when 'aix'
+  describe package('bos.rte') do
+    it { should be_installed }
+    its('version') { should match /^(6|7)\.1[.|\d]+\d/ }
```
Comment (chris-rock) [inline test/integration/test/integration/default/package_spec.rb]: I think that should be `its('version') { should match /^(6|7)[.|\d]+\d/ }`. Then it works out of the box with AIX 7.2 beta

File: lib/resources/port.rb [hunk 2]
```diff
@@ -34,8 +34,14 @@ def initialize(ip = nil, port) # rubocop:disable OptionalArguments
     case inspec.os[:family]
     when 'ubuntu', 'debian', 'redhat', 'fedora', 'centos', 'arch', 'wrlinux'
       @port_manager = LinuxPorts.new(inspec)
-    when 'darwin'
-      @port_manager = DarwinPorts.new(inspec)
+    when 'darwin', 'aix'
+      # AIX: see http://www.ibm.com/developerworks/aix/library/au-lsof.html#resources
+      #      and https://www-01.ibm.com/marketing/iwm/iwm/web/reg/pick.do?source=aixbp
```
Comment (chris-rock) [inline lib/resources/port.rb:39]: I cannot find `lsof` on my standard PDP AIX 7.2 and 6.1 machine. Should `lsof` be available on all machines?

If I run this locally I get

```
l490vp002_pub[/home/u0016746] > lsof
ksh: lsof:  not found
```

**Comment by chris-rock** [comment]
Hi @foobarbam thanks again for this great addition. Some questions:
- How have you prepared the AIX machine? I think we should automate the configuration with chef cookbooks. What do you think?
- I have troubles with `lsof` on my PDP AIX machines. Is this a standard command we can expect on all AIX machines?
- I am missing unit tests for the AIX operating system. We should test the parsing on a unit level, to ensure nobody will break the AIX features in future. Those are located at https://github.com/chef/inspec/tree/master/test/unit/resources
- Could you please rebase on `master`
- Some commits have the same title, can we squash some of the commits

Here are my failure reports:
- AIX 7.2 beta https://gist.github.com/chris-rock/bda25ab266c90e6c2455
- AIX 6.1 https://gist.github.com/chris-rock/23d6a120d07c553ffdc7

Please let me know if you need help.

File: lib/resources/port.rb [hunk 3]
```diff
@@ -129,42 +135,102 @@ def info
...
+    #      'ipv6:tcp:*',
+    #   ],
+    #   '456:ntpd' => [
+    #      'ipv4:udp:123:*',
+    #      'ipv6:udp:123:*',
+    #   ]
+    # }
+    proc_id = port_id = nil
+    lsof_cmd.stdout.each_line do |line|
+      line.chomp!
```
Comment (chris-rock) [inline lib/resources/port.rb:191]: Can we extract the line parser in a separate function?

File: lib/resources/port.rb [hunk 4]
```diff
@@ -129,42 +135,102 @@ def info
...
+
+  def initialize(inspec, lsofpath = nil)
+    @lsof = lsofpath || 'lsof'
+    super(inspec)
+  end
 
+  # rubocop:disable Metrics/CyclomaticComplexity
+  # rubocop:disable Metrics/PerceivedComplexity
+  # rubocop:disable Metrics/AbcSize
+  # rubocop:disable MethodLength
```
Comment (chris-rock) [inline lib/resources/port.rb]: We should try not to disable the rubocup checks. I think we can cover most recommendations, once we extract the line parsing.

File: Rakefile [hunk 5]
```diff
@@ -46,6 +46,21 @@ namespace :test do
     path = File.join(File.dirname(__FILE__), 'test', 'integration')
     sh('sh', '-c', "cd #{path} && bundle exec kitchen test -c #{concurrency} -t .")
   end
+
+  task :ssh, [:target] do |_t, args|
+    tests_path = File.join(File.dirname(__FILE__), 'test', 'integration', 'test', 'integration', 'default')
+    key_files = ENV['key_files'] || File.join(ENV['HOME'], '.ssh', 'id_rsa')
+
+    #sh_cmd =  "cd #{path} && target=#{args[:target]} key_files=#{key_files}"
```
Comment (chris-rock) [inline Rakefile]: please remove the comment

File: test/unit/mock/cmd/lsof-nP-i-FpctPn [hunk 6]
```diff
@@ -0,0 +1,63 @@
+p697
```
Comment (chris-rock) [inline test/unit/mock/cmd/lsof-nP-i-FpctPn:1]: can we reduce this file to the checked services

File: test/helper.rb [hunk 7]
```diff
@@ -135,6 +135,8 @@ def md.directory?
       '/sbin/sysctl -q -n net.ipv4.conf.all.forwarding' => cmd.call('sbin_sysctl'),
       # ports on windows
       'Get-NetTCPConnection | Select-Object -Property State, Caption, Description, LocalAddress, LocalPort, RemoteAddress, RemotePort, DisplayName, Status | ConvertTo-Json' => cmd.call('get-net-tcpconnection'),
+      # lsof formatted list of ports (should quite cross platform)
+      'lsof -nP -i -FpctPn' => cmd.call('lsof-nP-i-FpctPn'),
       # ports on mac
       'lsof -nP -iTCP -iUDP -sTCP:LISTEN' => cmd.call('lsof-np-itcp'),
```
Comment (chris-rock) [inline test/helper.rb]: as far as I can see this, we do not need the old lsof command anymore after your addition. Could you remove this please?

File: lib/resources/port.rb (hunk 2 above)
Comment (foobarbam) [inline lib/resources/port.rb:39]: there's a link in the code to where lsof can be downloaded from.  I know it is not standard AIX per se, but honestly there's no good way of doing what we're trying to do here on AIX without it.

File: lib/resources/port.rb (hunk 3 above)
Comment (foobarbam) [inline lib/resources/port.rb:191]: of course

**Comment by foobarbam** [comment]
sorry guys, missed most of these comments due to taking the holidays off.  I'll be back next week to address them all.

Thanks for the feedback!

File: lib/resources/port.rb (hunk 2 above)
Comment (chris-rock) [inline lib/resources/port.rb:39]: agreed, lets just document it and add and installation method to the integration tests

File: lib/resources/port.rb (hunk 3 above)
Comment (foobarbam) [inline lib/resources/port.rb:191]: done

File: test/unit/mock/cmd/lsof-nP-i-FpctPn (hunk 6 above)
Comment (foobarbam) [inline test/unit/mock/cmd/lsof-nP-i-FpctPn:1]: done

File: test/integration/test/integration/default/package_spec.rb (hunk 1 above)
Comment (foobarbam) [inline test/integration/test/integration/default/package_spec.rb]: done

File: lib/resources/port.rb (hunk 4 above)
Comment (foobarbam) [inline lib/resources/port.rb]: done

File: Rakefile (hunk 5 above)
Comment (foobarbam) [inline Rakefile]: done

**Comment by chris-rock** [comment]
awesome work @foobarbam thanks for adding this great AIX support

**Pull request merged**